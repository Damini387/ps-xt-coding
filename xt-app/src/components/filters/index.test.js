import React from 'react';
import { mount } from 'enzyme';

import { Provider } from 'react-redux';

import Filters from './index';

import { createStore, combineReducers } from "redux";

import getAllData from "../../store/reducers";

const rootReducer = combineReducers({
  getAllData
});

const store = createStore(rootReducer);

let wrapper = mount(
	<Provider store={ store }>
		<Filters />
	</Provider> );

describe('Filters', () => {

	it('should render the Filters Component correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should render years, success launch and success landing components', () => {
		expect(wrapper.find('.launchYear .heading').text()).toEqual("Launch Year");
        expect(wrapper.find('.successLaunch .heading').text()).toEqual("Successful Launch");
        expect(wrapper.find('.successLanding .heading').text()).toEqual("Successful Landing");
	});

});