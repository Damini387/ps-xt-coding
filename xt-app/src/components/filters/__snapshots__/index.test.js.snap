// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render the Filters Component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Filters)>
    <Filters
      data={Array []}
      dispatch={[Function]}
    >
      <div
        className="filters"
      >
        <h3>
          Filters
        </h3>
        <div
          className="launchYear"
        >
          <div
            className="heading"
          >
            Launch Year
          </div>
          <hr />
          <div
            className="allYears"
          >
            <Button
              click={[Function]}
              disabled={true}
              key="0"
              value="2006"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2006
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="1"
              value="2007"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2007
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="2"
              value="2008"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2008
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="3"
              value="2009"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2009
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="4"
              value="2010"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2010
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="5"
              value="2011"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2011
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="6"
              value="2012"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2012
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="7"
              value="2013"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2013
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="8"
              value="2014"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2014
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="9"
              value="2015"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2015
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="10"
              value="2016"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2016
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="11"
              value="2017"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2017
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="12"
              value="2018"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2018
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="13"
              value="2019"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2019
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              key="14"
              value="2020"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                2020
              </button>
            </Button>
          </div>
        </div>
        <div
          className="successLaunch"
        >
          <div
            className="heading"
          >
            Successful Launch
          </div>
          <hr />
          <div
            className="buttons"
          >
            <Button
              click={[Function]}
              disabled={true}
              value="True"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                True
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              value="False"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                False
              </button>
            </Button>
          </div>
        </div>
        <div
          className="successLanding"
        >
          <div
            className="heading"
          >
            Successful Landing
          </div>
          <hr />
          <div
            className="buttons"
          >
            <Button
              click={[Function]}
              disabled={true}
              value="True"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                True
              </button>
            </Button>
            <Button
              click={[Function]}
              disabled={true}
              value="False"
            >
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
              >
                False
              </button>
            </Button>
          </div>
        </div>
      </div>
    </Filters>
  </Connect(Filters)>
</Provider>
`;
